### PR TITLE
Diagram zoom and pan

### PR DESCRIPTION
**Pull request by areksvk**
Fix small radius curves

### CODE REVIEW

File: src/Component.js [hunk 1]
```diff
@@ -103,9 +103,9 @@ class Component {
         if (!this.isRendered()) {
             throw "Can't call setData() when component is not rendered, please call .render() first.";
         }
-        this._setData(this._container, data, this._lastData);
         this._lastData = Object.assign({}, this._lastData, data);
-        return this;
+
+        return this._setData(this._container, data, this._lastData);
```
Comment (mattesCZ) [inline src/Component.js]: I'm not getting why we're passing `this._lastData` as the third argument. Isn't it some relic?

File: src/Diagram.js [hunk 2]
```diff
@@ -104,18 +117,65 @@ class Diagram extends Component {
...
+        this._zoomOutScaleHeight = this._getZoomOutScale(svgDimensions.height, maxHeight);
+
+        this._zoomOutScale = Math.min(this._zoomOutScaleWidth, this._zoomOutScaleHeight);
+
+        this._zoom = d3.zoom()
+            .extent([[0, 0], [svgDimensions.width, svgDimensions.height]])
+            .scaleExtent([this._zoomOutScale, 1])
+            .translateExtent([[0, 0], [maxWidth, maxHeight]]);
+
+        this._zoom.on("zoom", this._zoomHandler.bind(this));
```
Comment (mattesCZ) [inline src/Diagram.js]: Isn't it necessary to unbind the event handler?

**Comment by mattesCZ** [comment]
Thanks for the updates, approve!

**Comment by areksvk** [comment]
I cancel this pull request and create a new one, rebased on new master with higher version

**Pull request closed without merging**